bee-ledger/src/workers/snapshot/download.rs

```diff
@@ -1,37 +1,89 @@
 // Copyright 2020-2021 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::workers::snapshot::error::Error;
+use crate::{types::snapshot::SnapshotHeader, workers::snapshot::error::Error};
 
+use bee_common::packable::Packable;
+
+use bytes::Buf;
+use futures::{future::join_all, StreamExt};
 use log::{info, warn};
 
-use std::path::Path;
+use std::{io::Read, path::Path};
+
+async fn download_snapshot_header(download_url: &str) -> Result<Option<(SnapshotHeader, &str)>, Error> {
+    info!("Downloading snapshot header {}...", download_url);
+
+    match reqwest::get(download_url).await {
+        Ok(res) => {
+            let mut stream = res.bytes_stream();
+            let mut bytes = Vec::<u8>::new();
```

Comment:
Could it be worth it to `Vec::with_capacity(SnapshotHeader::LENGTH)`?

Reply:
Yes, I think so.